pangeo_forge_runner/bakery/flink.py

```diff
@@ -150,13 +150,101 @@ class FlinkOperatorBakery(Bakery):
+        https://github.com/pangeo-forge/pangeo-forge-cloud-federation/blob/main/terraform/aws/operator.tf
+        
+        If using that Terraform you can configure the path via `historyserver_mount_path`:
+        https://github.com/pangeo-forge/pangeo-forge-cloud-federation/blob/main/terraform/aws/variables.tf
+        """,
+    )
+
+    def add_job_manager_pod_template(self, current_flink_deploy: dict):
+        """Add the job manager pod template to the existing flink deploy
+
+        Returns:
+            a dictionary representing the whole `kind: flinkdeployment`
+        """
+        pod_template = {
+            "podTemplate": {
+                "spec": {
+                    "securityContext": {
+                        # flink uid/guid
+                        "fsGroup": 9999
+                    },
+                    "initContainers": [
+                        {
+                            "name": "efs-mount-ownership-fix",
+                            "image": "busybox:1.36.1",
+                            "command": [
+                                "sh",
+                                "-c",
+                                # makes sure the flink uid/gid is owner of the archival mount
+                                f"chown 9999:9999 {self.job_archive_efs_mount} && ls -lhd {self.job_archive_efs_mount}",
+                            ],
+                            "volumeMounts": [
+                                {
+                                    "name": "efs-flink-history",
+                                    "mountPath": f"{self.job_archive_efs_mount}",
+                                }
+                            ],
+                        }
+                    ],
+                    "containers": [
+                        {
+                            # NOTE: "name" and "image" are required here
+                            # and were taken from existing deployed manifests
+                            # all other attributes get back-filled in by the operator
+                            "name": "flink-main-container",
+                            "image": f"flink:{self.flink_version}",
+                            "volumeMounts": [
+                                {
+                                    "name": "efs-flink-history",
+                                    "mountPath": f"{self.job_archive_efs_mount}",
+                                }
+                            ],
+                        }
+                    ],
+                    "volumes": [
+                        {
+                            "name": "efs-flink-history",
+                            "persistentVolumeClaim": {
+                                "claimName": "flink-historyserver-efs-pvc"
+                            },
+                        }
+                    ],
+                }
+            },
+        }
+
+        # shallow copy
+        new_flink_deploy = current_flink_deploy.copy()
```

Comment:
Because this struct has dicts inside it, we need a deep copy here with https://docs.python.org/3/library/copy.html#copy.deepcopy.

Reply:
doh, I should've caught that one. donseo